### PR TITLE
imp(08-wasm): removed panic in `GetCodeHash` and `AddCodeHash`

### PR DESCRIPTION
**Pull request by srdtrk**
## Description

closes: https://github.com/cosmos/ibc-go/issues/4947

### Commit Message / Changelog Entry

```text
imp(08-wasm): removed panic in `GetCodeHash` and `AddCodeHash`
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/light-clients/08-wasm/keeper/genesis_test.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func (suite *KeeperTestSuite) TestInitGenesis() {
 			suite.Require().NoError(err)
 
 			var storedHashes []string
-			res := types.GetCodeHashes(suite.chainA.GetContext(), GetSimApp(suite.chainA).AppCodec())
+			res, err := types.GetCodeHashes(suite.chainA.GetContext(), GetSimApp(suite.chainA).AppCodec())
```
Comment (damiannolan) [inline modules/light-clients/08-wasm/keeper/genesis_test.go:60]: should we move `suite.Require().NoError(err)` from L65 up to check the error here? 

guess it doesn't make difference actually!

File: modules/light-clients/08-wasm/keeper/grpc_query.go [hunk 2]
```diff
@@ -46,7 +46,10 @@ func (k Keeper) CodeHashes(goCtx context.Context, req *types.QueryCodeHashesRequ
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
 	var codeHashes []string
-	storedHashes := types.GetCodeHashes(ctx, k.cdc)
+	storedHashes, err := types.GetCodeHashes(ctx, k.cdc)
+	if err != nil {
+		return nil, status.Error(codes.DataLoss, err.Error())
+	}
 	for _, hash := range storedHashes.Hashes {
```
Comment (damiannolan) [inline modules/light-clients/08-wasm/keeper/grpc_query.go:53]: CR for breathing room maybe? :) 

```suggestion
	}
	
	for _, hash := range storedHashes.Hashes {
```

File: modules/light-clients/08-wasm/types/wasm.go [hunk 3]
```diff
@@ -11,32 +11,49 @@ import (
...
-func GetCodeHashes(ctx sdk.Context, cdc codec.BinaryCodec) CodeHashes {
+func GetCodeHashes(ctx sdk.Context, cdc codec.BinaryCodec) (CodeHashes, error) {
 	wasmStoreKey := ibcwasm.GetWasmStoreKey()
 	store := ctx.KVStore(wasmStoreKey)
 	bz := store.Get([]byte(KeyCodeHashes))
 	if len(bz) == 0 {
-		return CodeHashes{}
+		return CodeHashes{}, nil
 	}
 	var hashes CodeHashes
```
Comment (damiannolan) [inline modules/light-clients/08-wasm/types/wasm.go:21]: maybe here for good measure too! 

```suggestion
	}
	
	var hashes CodeHashes
```

File: modules/light-clients/08-wasm/types/wasm.go [hunk 4]
```diff
@@ -11,32 +11,49 @@ import (
...
-	cdc.MustUnmarshal(bz, &hashes)
-	return hashes
+	err := cdc.Unmarshal(bz, &hashes)
+	if err != nil {
+		return CodeHashes{}, err
+	}
+	return hashes, nil
```
Comment (damiannolan) [inline modules/light-clients/08-wasm/types/wasm.go:26]: last one I swear 

```suggestion
	}
	
	return hashes, nil
```

**Review by charleenfei — APPROVED** [review]
thank you! just one comment about a test case

**Review by DimitrisJim — APPROVED** [review]
lvgtm!

File: modules/light-clients/08-wasm/types/wasm_test.go [hunk 5]
```diff
@@ -41,7 +42,8 @@ func (suite *TypesTestSuite) TestGetCodeHashes() {
 
 			tc.malleate()
 
-			codeHashes := types.GetCodeHashes(suite.chainA.GetContext(), GetSimApp(suite.chainA).AppCodec())
```
Comment (charleenfei) [inline modules/light-clients/08-wasm/types/wasm_test.go:44]: sorry, i know this wasn't in your PR but also, do we need `tc := tc` above on line 38?

Comment (srdtrk) [reply modules/light-clients/08-wasm/types/wasm_test.go:44]: Not sure, I'm in favour of leaving it as is. Doesn't hurt anything

**Review by crodriguezvega — COMMENTED** [review]
Good catch, and thank you for fixing it, @srdtrk.

**Pull request merged**